src/pocketmine/Player.php

```diff
@@ -2241,7 +2242,34 @@ public function handleInventoryTransaction(InventoryTransactionPacket $packet) :
 
 				switch($type){
 					case InventoryTransactionPacket::USE_ITEM_ON_ENTITY_ACTION_INTERACT:
-						break; //TODO
+						if(!$target->isAlive()){
+							return true;
+						}
+
+						$heldItem = $this->inventory->getItemInHand();
+
+						$ev = new PlayerGentlyTouchEntityEvent($this, $target, $heldItem, $packet->trData->clickPos);
+
+						if(!$this->canInteract($target, 8)){
+							$ev->setCancelled();
+						}
+
+						$this->server->getPluginManager()->callEvent($ev);
+
+						$target->interact($ev);
```

Comment:
what exactly is the point of passing the event here? Seems like extra work for no good reason.

Reply:
Entity::attack() calls the event passed to it. Here you're calling it externally and then passing it to the hook, which seems pointless.